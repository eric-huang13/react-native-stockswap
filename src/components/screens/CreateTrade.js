import React, {Component} from 'react';
import {
  Text,
  View,
  StyleSheet,
  TextInput,
  SafeAreaView,
  KeyboardAvoidingView,
  Keyboard,
  TouchableWithoutFeedback,
  TouchableOpacity,
  ScrollView
} from 'react-native';
import TriangleIcon from '../../icons/TriangleIcon';
import {Formik} from 'formik';
import * as Yup from 'yup';
import { moderateScale } from '../../util/responsiveFont';

const validationSchema = Yup.object().shape({
  // Validation will change when we have a better idea what data needs to be sent
  name: Yup.string()
    .label('name')
    .required('Name is required')
    .min(2, 'Name must have more than 2 characters '),

  time: Yup.number()
    .label('time')
    .typeError('Please enter a number')
    .required('Time is required')
    .min(2, 'Time must have more than 2 characters '),

  quantity: Yup.number()
    .typeError('Please enter a number')
    .label('quantity')
    .required('Quantity is required')
});

export default class CreateTrade extends Component {
  constructor(props) {
    super(props);

    this.state = {
      isPrivate:false,
      shouldShow: false,
    };
  }

  dropDownSelect(setting) {
    this.setState({isPrivate: setting, shouldShow: false});
  }

  render() {
    const {shouldShow} = this.state;

    return (
      <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
        <SafeAreaView style={style.mainContainer}>
          <ScrollView>
          <KeyboardAvoidingView>
          <Formik
                initialValues={{
                name: '',
                orderType: 'buy',
                time: '',
                quantity: '',
                isPrivate:false,                                 
                }}
                onSubmit={(values) => {
                  console.log(values, 'values');
                  // UserPost(values)
                }}
                validationSchema={validationSchema}>
                {({
                  handleChange,
                  values,
                  handleSubmit,
                  errors,
                  isValid,
                  touched,
                  handleBlur,
                  isSubmitting,
                  setFieldValue,
                }) => (
                  <View>
            <Text style={style.header}>Post a Trade</Text>
            <View style={style.uploadContainer}>
              <Text style={style.uploadText}>
                Upload trade screenshot for autofill
              </Text>
            </View>
            <View style={style.nameInputContainer}>
              <Text style={style.inputHeader}>Stock name</Text>
              <TextInput
                onBlur={handleBlur('name')}
                value={values.name}
                onChangeText={handleChange('name')}              
                style={style.inputStyleBody}                
                placeholder="Start typing stock name"
                placeholderTextColor="#9ea6b5"                               
              />            
              <Text style={style.errorText}>
                          {touched.name && errors.name}
                        </Text>
            </View>
            <View>
              <Text style={style.inputHeader}>Order Type</Text>
              <View style={style.buttonsContainer}>
                <TouchableOpacity onPress={() => {
                                    setFieldValue('orderType', 'buy');
                                  }}>
                  <Text style={
                  values.orderType === 'buy'
                    ? {...style.buyButton, backgroundColor: '#8b64ff'}
                    : {...style.buyButton}
                }>Buy</Text>
                </TouchableOpacity>
                <TouchableOpacity onPress={() => setFieldValue('orderType', 'sell')}>
                  <Text style={values.orderType === 'sell'
                    ? {...style.buyButton, backgroundColor: '#8b64ff'}
                    : {...style.buyButton}}>Sell</Text>
                </TouchableOpacity>
              </View>
            </View>
            <View style={style.TimeContainer}>
              <Text style={style.inputHeader}>Time</Text>
              <TextInput
                onBlur={handleBlur('time')}
                value={values.time}
                onChangeText={handleChange('time')} 
                style={style.inputStyleBody}                
                placeholder="Time when you sell it"
                placeholderTextColor="#9ea6b5"
              />
              <Text style={style.errorText}>
                          {touched.time && errors.time}
                        </Text>
            </View>
            <View style={style.quantityConatiner}>
              <Text style={style.inputHeader}>Quantity</Text>
              <TextInput
                onBlur={handleBlur('quantity')}
                value={values.quantity}
                onChangeText={handleChange('quantity')} 
                style={style.inputStyleBody}               
                placeholder="Number of shares"
                placeholderTextColor="#9ea6b5"                
              />
              <Text style={style.errorText}>
                          {touched.quantity && errors.quantity}
                        </Text>
            </View>
            <View>
              <Text style={style.inputHeader}>Stock privacy</Text>


              <View style={style.dotsDropdownContainer}>
                          <TouchableOpacity
                            onPress={() =>
                              this.setState({
                                shouldShow: !shouldShow,
                              })
                            }>
                            <View style={style.visibleButtonContainer}>
                              <Text style={style.middleDetailsText}>
                                {this.state.isPrivate == false ? 
                                "Visible for all"
                                :
                                "Private"  
                              }
                              </Text>
                              <TriangleIcon style={style.icon} />
                            </View>
                          </TouchableOpacity>
                          {this.state.shouldShow ? (
                            <View style={style.dropdown}>
                              {this.state.isPrivate == false ? (
                                <TouchableOpacity
                                  onPress={() => {
                                    this.dropDownSelect(true);
                                    setFieldValue('isPrivate', true);
                                  }}>
                                  <Text style={style.dropDownText}>
                                    Private
                                  </Text>
                                </TouchableOpacity>
                              ) : (
                                <TouchableOpacity
                                  onPress={() => {
                                    this.dropDownSelect(false);
                                    setFieldValue('isPrivate', false);
                                  }}>
                                  <Text style={style.dropDownText}>
                                    Visible for all
                                  </Text>
                                </TouchableOpacity>
                              )}
                            </View>
                          ) : null}
                        </View>
            </View>
            <TouchableOpacity onPress={handleSubmit}>
            <Text style={style.publishButton}>Publish</Text>
            </TouchableOpacity>
            </View>

            )}
            </Formik>
          </KeyboardAvoidingView>
          </ScrollView>
        </SafeAreaView>
      </TouchableWithoutFeedback>
    );
  }
}

const style = StyleSheet.create({
  mainContainer: {
    flex: 1,
    backgroundColor: '#2a334a',
    paddingVertical: moderateScale(8),
    paddingHorizontal: moderateScale(36),
    flexDirection: 'column',
    justifyContent: 'space-around',
  },
  header: {
    fontSize: moderateScale(20),
    fontFamily: 'Montserrat-Bold',
    color: '#FFFFFF',
    textAlign: 'center',
    marginBottom: moderateScale(12),
  },
  uploadContainer: {
    backgroundColor: '#46486e',
    height: moderateScale(48),
    borderRadius: 2,
    alignItems: 'center',
    justifyContent: 'center',
  },

  uploadText: {
    color: '#FFFFFF',
    fontFamily: 'Montserrat-Regular',
    fontSize: moderateScale(14),
  },
  nameInputContainer: {
    marginTop: moderateScale(20),
  },
  inputHeader: {
    fontSize: moderateScale(12),
    color: '#babec8',
    fontFamily: 'Montserrat-Regular',
    marginBottom: moderateScale(3),
  },
  inputStyleBody: {
    borderRadius: 6,
    backgroundColor: '#3e4d6c',
    padding: moderateScale(8),
    marginTop: moderateScale(1),
    fontSize: moderateScale(14),
    textAlignVertical: 'center',
    fontFamily: 'Montserrat-Italic',
    color: '#FFFFFF',
    height: moderateScale(42),

      },
  buttonsContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginBottom:moderateScale(18),
    marginTop:moderateScale(2)

      },
  buyButton: {
    alignSelf: 'center',
    color: '#FFFFFF',
    textAlign: 'center',
    paddingVertical: moderateScale(12),
    paddingHorizontal: moderateScale(20),
    width: moderateScale(141),
    borderRadius: 6,
    fontSize: moderateScale(14),
    fontFamily: 'Montserrat-SemiBold',
    backgroundColor: '#536183',
  },
  sellButton: {
    alignSelf: 'center',
    backgroundColor: '#536183',
    color: '#FFFFFF',
    textAlign: 'center',
    paddingVertical: moderateScale(12),
    paddingHorizontal: moderateScale(20),
    width: moderateScale(141),
    borderRadius: 6,
    fontSize: moderateScale(14),
    fontFamily: 'Montserrat-SemiBold',
  },
  visibleButtonContainer: {
    paddingHorizontal: moderateScale(10),
    paddingVertical: moderateScale(10),
    borderRadius: 6,
    backgroundColor: '#3E4D6C',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  middleDetailsText: {
    fontFamily: 'Montserrat-Medium',
    color: '#FFFFFF',
    fontSize: moderateScale(16),
  },
  dropdown: {
    flexDirection: 'column',
    justifyContent: 'space-between',
    width: '100%',
    marginTop: moderateScale(41),
    backgroundColor: '#3E4D6C',
    zIndex: 1,
    paddingVertical: moderateScale(4),
    height: moderateScale(35),
    position: 'absolute',
    borderBottomLeftRadius: 6,
    borderBottomRightRadius: 6,
  },
  dropDownText: {
    color: 'white',
    fontSize: moderateScale(16),
    marginHorizontal: moderateScale(12),
    fontFamily: 'Montserrat-Medium',
    marginBottom: moderateScale(6),
  },
  dropDownTextReportContainer: {
    borderTopWidth: 1,
    borderTopColor: 'lightgrey',
    paddingTop: moderateScale(4),
    backgroundColor: '#2C3957',
  },
  icon: {
    marginRight: moderateScale(4),
  },
  publishButton: {
    alignSelf: 'center',
    backgroundColor: '#8b64ff',
    color: '#FFFFFF',
    textAlign: 'center',
    paddingVertical: moderateScale(12),
    paddingHorizontal: moderateScale(20),
    width: moderateScale(303),
    borderRadius: 6,
    fontSize: moderateScale(14),
    marginTop: moderateScale(36),
    fontFamily: 'Montserrat-SemiBold',
    marginBottom: moderateScale(8),

  },
  errorText: {
    color: '#F66E6E',
    fontWeight: 'bold',
    textAlign: 'center',
  },
});
