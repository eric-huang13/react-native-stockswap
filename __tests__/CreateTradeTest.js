import React from 'react';
import CreateTrade from '../src/components/PostTabScreens/CreateTrade';
import renderer from 'react-test-renderer';


jest.mock("@react-native-async-storage/async-storage", () =>
  require("@react-native-async-storage/async-storage/jest/async-storage-mock"),
);
jest.mock('react-redux', () => ({
    connect: () => {
      return (component) => {
        return component
      };
    }
  }));
  

test('renders correctly', () => {
    const tree = renderer.create(<CreateTrade/>).toJSON();
  expect(tree).toMatchSnapshot();
});