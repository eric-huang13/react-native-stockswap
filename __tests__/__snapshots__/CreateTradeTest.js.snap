// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  accessible={true}
  emulateUnlessSupported={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#2a334a",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "space-around",
      "paddingHorizontal": 50.846715328467155,
      "paddingVertical": 11.299270072992702,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        onLayout={[Function]}
      >
        <View>
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Montserrat-Bold",
                "fontSize": 28.248175182481752,
                "marginBottom": 16.94890510948905,
                "textAlign": "center",
              }
            }
          >
            Post a Trade
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#46486e",
                "borderRadius": 2,
                "height": 67.7956204379562,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "Montserrat-Regular",
                  "fontSize": 19.77372262773723,
                }
              }
            >
              Upload trade screenshot for autofill
            </Text>
          </View>
          <View
            style={
              Object {
                "marginTop": 28.248175182481752,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#babec8",
                  "fontFamily": "Montserrat-Regular",
                  "fontSize": 16.94890510948905,
                  "marginBottom": 4.237226277372263,
                }
              }
            >
              Stock name
            </Text>
            <TextInput
              allowFontScaling={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              placeholder="Start typing stock name"
              placeholderTextColor="#9ea6b5"
              rejectResponderTermination={true}
              style={
                Object {
                  "backgroundColor": "#3e4d6c",
                  "borderRadius": 6,
                  "color": "#FFFFFF",
                  "fontFamily": "Montserrat-Italic",
                  "fontSize": 19.77372262773723,
                  "height": 59.32116788321168,
                  "marginTop": 1.4124087591240877,
                  "padding": 11.299270072992702,
                  "textAlignVertical": "center",
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />
            <Text
              style={
                Object {
                  "color": "#F66E6E",
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            />
          </View>
          <View>
            <Text
              style={
                Object {
                  "color": "#babec8",
                  "fontFamily": "Montserrat-Regular",
                  "fontSize": 16.94890510948905,
                  "marginBottom": 4.237226277372263,
                }
              }
            >
              Order Type
            </Text>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 25.423357664233578,
                  "marginTop": 2.8248175182481754,
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "alignSelf": "center",
                      "backgroundColor": "#8b64ff",
                      "borderRadius": 6,
                      "color": "#FFFFFF",
                      "fontFamily": "Montserrat-SemiBold",
                      "fontSize": 19.77372262773723,
                      "paddingHorizontal": 28.248175182481752,
                      "paddingVertical": 16.94890510948905,
                      "textAlign": "center",
                      "width": 199.14963503649636,
                    }
                  }
                >
                  Buy
                </Text>
              </View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "alignSelf": "center",
                      "backgroundColor": "#536183",
                      "borderRadius": 6,
                      "color": "#FFFFFF",
                      "fontFamily": "Montserrat-SemiBold",
                      "fontSize": 19.77372262773723,
                      "paddingHorizontal": 28.248175182481752,
                      "paddingVertical": 16.94890510948905,
                      "textAlign": "center",
                      "width": 199.14963503649636,
                    }
                  }
                >
                  Sell
                </Text>
              </View>
            </View>
          </View>
          <View>
            <Text
              style={
                Object {
                  "color": "#babec8",
                  "fontFamily": "Montserrat-Regular",
                  "fontSize": 16.94890510948905,
                  "marginBottom": 4.237226277372263,
                }
              }
            >
              Time
            </Text>
            <TextInput
              allowFontScaling={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              placeholder="Time when you sell it"
              placeholderTextColor="#9ea6b5"
              rejectResponderTermination={true}
              style={
                Object {
                  "backgroundColor": "#3e4d6c",
                  "borderRadius": 6,
                  "color": "#FFFFFF",
                  "fontFamily": "Montserrat-Italic",
                  "fontSize": 19.77372262773723,
                  "height": 59.32116788321168,
                  "marginTop": 1.4124087591240877,
                  "padding": 11.299270072992702,
                  "textAlignVertical": "center",
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />
            <Text
              style={
                Object {
                  "color": "#F66E6E",
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            />
          </View>
          <View>
            <Text
              style={
                Object {
                  "color": "#babec8",
                  "fontFamily": "Montserrat-Regular",
                  "fontSize": 16.94890510948905,
                  "marginBottom": 4.237226277372263,
                }
              }
            >
              Quantity
            </Text>
            <TextInput
              allowFontScaling={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              placeholder="Number of shares"
              placeholderTextColor="#9ea6b5"
              rejectResponderTermination={true}
              style={
                Object {
                  "backgroundColor": "#3e4d6c",
                  "borderRadius": 6,
                  "color": "#FFFFFF",
                  "fontFamily": "Montserrat-Italic",
                  "fontSize": 19.77372262773723,
                  "height": 59.32116788321168,
                  "marginTop": 1.4124087591240877,
                  "padding": 11.299270072992702,
                  "textAlignVertical": "center",
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />
            <Text
              style={
                Object {
                  "color": "#F66E6E",
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            />
          </View>
          <View>
            <Text
              style={
                Object {
                  "color": "#babec8",
                  "fontFamily": "Montserrat-Regular",
                  "fontSize": 16.94890510948905,
                  "marginBottom": 4.237226277372263,
                }
              }
            >
              Stock privacy
            </Text>
            <View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#3E4D6C",
                      "borderRadius": 6,
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "paddingHorizontal": 14.124087591240876,
                      "paddingVertical": 14.124087591240876,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#FFFFFF",
                        "fontFamily": "Montserrat-Medium",
                        "fontSize": 22.598540145985403,
                      }
                    }
                  >
                    Visible for all
                  </Text>
                  <RNSVGSvgView
                    align="xMidYMid"
                    bbHeight={12.711678832116789}
                    bbWidth={16.94890510948905}
                    focusable={false}
                    height={12.711678832116789}
                    meetOrSlice={0}
                    minX={0}
                    minY={0}
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "borderWidth": 0,
                        },
                        Object {
                          "marginRight": 5.649635036496351,
                        },
                        Object {
                          "flex": 0,
                          "height": 12,
                          "width": 16,
                        },
                      ]
                    }
                    vbHeight={9}
                    vbWidth={12}
                    width={16.94890510948905}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <RNSVGGroup>
                      <RNSVGPath
                        d="M6.79 7.985l4.955-6.371A1 1 0 0010.955 0h-9.91a1 1 0 00-.79 1.614l4.956 6.371a1 1 0 001.578 0z"
                        fill={4290289919}
                        fillRule={0}
                        propList={
                          Array [
                            "fill",
                            "fillRule",
                          ]
                        }
                      />
                    </RNSVGGroup>
                  </RNSVGSvgView>
                </View>
              </View>
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "#8b64ff",
                  "borderRadius": 6,
                  "color": "#FFFFFF",
                  "fontFamily": "Montserrat-SemiBold",
                  "fontSize": 19.77372262773723,
                  "marginBottom": 11.299270072992702,
                  "marginTop": 50.846715328467155,
                  "paddingHorizontal": 28.248175182481752,
                  "paddingVertical": 16.94890510948905,
                  "textAlign": "center",
                  "width": 427.95985401459853,
                }
              }
            >
              Publish
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
